src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/NotifyListeners.kt

```diff
@@ -0,0 +1,29 @@
+// ktlint-disable filename
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Component
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.NotifyService
+
+@Component
+class NotifyReferralEventListener(
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.sent-referral}") private val interventionsUISentReferralLocation: String,
+  private val notifyService: NotifyService,
+) : ApplicationListener<ReferralEvent> {
+  override fun onApplicationEvent(event: ReferralEvent) {
+    when (event.type) {
+      ReferralEventType.SENT -> {
```

Comment:
_musing, not blocking_
Checking: this is part of #66, right? (This PR could have a "base" on top of that branch, so it only shows additions, with the trade-off of some extra rebases -- or it could be split to "foundations" and 2 PRs adding listener logic?)

Reply:
this PR contains all the commits from the other one - the most recent commit contains just the new changes.

not sure how to isolate this. you can create a PR on the other branch, but that isn't want we want, it needs to go into main. so as soon as the other branch is merged this will become clearer.